v2/fthealth.go

```diff
@@ -0,0 +1,108 @@
+package v2
+
+import (
+	"sync"
+	"time"
+)
+
+type HealthCheck struct {
+	SystemCode  string
+	Name        string
+	Description string
+	Checks      []Check
+	Parallel    bool
+}
+
+type Check struct {
+	ID               string
+	Name             string
+	Severity         uint8
+	BusinessImpact   string
+	TechnicalSummary string
+	PanicGuide       string
+	Checker          func() (string, error)
+}
+
+func RunCheck(hc *HealthCheck) HealthResult {
+	return hc.health()
+}
+
+func (ch *HealthCheck) health() (result HealthResult) {
```

Comment:
Actually I agree with you, structs owning the methods that work on themselves would be the best. Thx